miqi/agent/tools/filesystem.py

```
@@ -462,6 +462,176 @@ def _get_session_workspace(base_workspace: Path | None, sandbox) -> Path | None:
+    Sandbox-aware when a WSL sandbox is active; otherwise a native
+    ``Path.exists()`` probe (Windows/posix).
+    """
+    if sandbox is not None and getattr(sandbox, "_use_wsl", False):
+
+        async def _check(p: str) -> bool:
+            try:
+                sb = _resolve_sandbox_path(
+                    p, session_ws, sandbox, extra_roots=shared_roots,
+                )
+                return await _sandbox_file_exists(sandbox, sb)
+            except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)